src/Schema/TableDiff.php

```diff
@@ -238,10 +251,44 @@ public function getAddedColumns(): array
         return array_values($this->addedColumns);
     }
 
-    /** @return list<ColumnDiff> */
+    /**
+     * @internal Use {@see getChangedColumns()} instead.
+     *
+     * @return list<ColumnDiff>
+     */
     public function getModifiedColumns(): array
     {
-        return array_values($this->changedColumns);
+        Deprecation::triggerIfCalledFromOutside(
+            'doctrine/dbal',
+            'https://github.com/doctrine/dbal/pull/6080',
+            '%s is deprecated, you should use `getModifiedColumns()` instead.',
+            __METHOD__,
+        );
+
+        return array_values(array_filter($this->getChangedColumns(), static function (ColumnDiff $diff) {
+            return count($diff->changedProperties) > 0;
+        }));
+    }
+
+    /** @return list<ColumnDiff> */
+    public function getChangedColumns(): array
+    {
+        if (! $this->wasRenamedInternalUsed && count($this->renamedColumns) > 0) {
+            Deprecation::trigger(
+                'doctrine/dbal',
+                'https://github.com/doctrine/dbal/pull/6080',
+                'Modifying $renamedColumns is deprecated, this property will be removed in the next major. ' .
+                'Set $modifiedColumns in the constructor instead',
+                __METHOD__,
+            );
+        }
+
+        $legacyChangedColumns = [];
+        foreach ($this->renamedColumns as $oldName => $column) {
+            $legacyChangedColumns[$oldName] = new ColumnDiff($oldName, $column, []);
+        }
+
+        return array_values(array_merge($this->changedColumns, $legacyChangedColumns));
```

Comment:
what if we already had a ColumnDiff for that column ?

Reply:
It's unlikely to happen as that would mean the user is modifying the `renamedColumn` internal property directly from an existing diff with changed columns (now all the dbal uses `changedColumns` only, I could handle the case and merge the diff though in the BC code I added a bit higher